packages/connectivity/src/scp-cf/cache.ts

```diff
@@ -1,18 +1,30 @@
+import { Destination } from './destination';
+
 interface CacheInterface<T> {
-  hasKey(key: string): boolean;
-  get(key: string): T | undefined;
-  set(key: string, item: T, expirationTime?: number): void;
+  get(key: string | undefined): Promise<T | undefined>;
+  set(key: string | undefined, item: CacheEntry<T>): Promise<void>;
+  clear(): Promise<void>;
+}
+
+/**
+ * Interface to implement custom destination caching.
+ */
+export interface DestinationCacheInterface extends CacheInterface<Destination> {
+  set(key: string | undefined, item: CacheEntry<Destination>): Promise<void>;
```

Comment:
[q] I assume you tried this but what happens if you use a conditional type like `type syncOrAsyng<isPromise extends boolean,U> = V extends true ? Promise<U> : U;` and return `syncOrAsyng<isPromise,void>`  for the set and add a second generic argument `isPromise` to the CacheInterface, perhaps `isAsync` is a better name.

Reply:
The destination cache would be async, but perhaps the other caches could remain sync this way.

Reply:
@florian-richter @jjtang1985 we also experimented with a conditional type which is either `Promise<T>` or `T` to get the cache for the sync and async case. However, this lead to an additional generic arguments and ugly return types. So I am not conviced what is better:
- having await where it is not needed
- havving a complicated type foo bla....